### ISSUE TITLE
[dask] eval_metric for dask sklearn wrapper

### ISSUE THREAD

**Issue by jborchma** [dmlc/xgboost#6121]
Playing around with the dask sklearn wrapper, I realized that the [`fit`-method](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/dask.py#L1073) currently does not support passing through `eval_metric` (as opposed to the normal [`fit`-method](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/sklearn.py#L442).

I was wondering: Is there a reason for this? And what would be involved in adding this feature in?

**Comment by trivialfis** [comment]
For builtin metrics inside XGBoost, normal parameter on constructor should suffice.  But for custom evaluation metric, synchronization is a bit problematic.  I'm hoping to resolve it along with early stopping support.

**Comment by mmccarty** [comment]
I think it would be ideal to have `eval_metric` as a kwarg to the fit method since its the sklearn convention and otherwise confuses users when they lean on API compatibility. For example, GridSearchCV normally looks like the following

```
from sklearn.model_selection import GridSearchCV
import xgboost as xgb

reg = xgb.XGBRegressor()

param_grid = {
    'max_depth': [4, 5],
    'min_child_weight': [10, 20],
    'learning_rate': [0.05]
}
scoring = {
    'rmse': lambda est, X, y: max(est.evals_result_["validation_0"]["rmse"])
}
gscv = GridSearchCV(reg, param_grid, scoring=scoring, error_score='raise', cv=2, refit='rmse', n_jobs=-1, verbose=1)
gscv.fit(X, y, eval_set=[(X, y)], eval_metric='rmse')
```
But with the dask interface as it currently stands, you have to remove `eval_metric` from fit and pass in when the estimator class is initialized.
```
reg = xgb.dask.DaskXGBRegressor(eval_metric='rmse')
gscv = GridSearchCV(reg, param_grid, scoring=scoring, error_score='raise', cv=2, n_jobs=1, refit='rmse', verbose=1)
gscv.fit(dX, dy, eval_set=[(dX, dy)])
```

cc @gforsyth @JohnZed

**Comment by trivialfis** [comment]
With the new call back interface and early stopping in, I believe this should not be far away.  Let me just get the K8S issue fixed first.

**Comment by trivialfis** [comment]
Closed by https://github.com/dmlc/xgboost/pull/6471
